lib/types.d.ts

```diff
@@ -0,0 +1,52 @@
+declare module 'json5' {
+  type StringifyOptions = Partial<{
```

Comment:
Instead of Partial<> you could mark every entry as optonal with a ?:

replacer?:  ...

Reply:
That is true, but the reason I used Partial is to avoid having to label each property with a question mark since they are all optional. Do you explicitly want me to use question marks?

Reply:
I have to agree that `Partial` _feels_ odd here, even though it achieves the same goal. For me, `Partial` is used when you expect to get a subset of data you normally would otherwise, like in a PATCH request for example. Specifying each property as options makes more sense from a documentation standpoint I think.

Reply:
Ok, no probs. I'll mark each property with a question mark. :+1:

Reply:
Just to return to this, `Partial` *does* label every property as optional. It's a [utility type](https://www.typescriptlang.org/docs/handbook/utility-types.html) which simply serves as a shortcut in making every property of an object optional which bypasses having to add a question mark to every one of them. I'll use `Partial` because I assume that every property of any object structured in the `StringifyOptions` mold will be an optional property.

Reply:
The use case for `Partial` is to transform an existing type into one that is a *partial* version of the original. Again, it `feels` odd here. I'd prefer to just be explicit here.